File: ClientApp/src/app/models/route.ts
//import { Pickup } from "./pickup";
import { CustomerPickup } from "./customerpickup";
import { routepricing } from "./routepricing";

export class Route {
    public id: number;
    public routeName: string;
    public routeDescription: string;
    public scheduledDeparture: string;
    public routeStarts: Date;
    public routeExpires: Date;
    public basePrice: number;
    public fromAddress: string;
    public fromGeoLat: number;
    public fromGeoLong: number;
    public toAddress: string;
    public toGeoLat: number;
    public toGeoLong: number;
    public routeDayMon: boolean;
    public routeDayTue: boolean;
    public routeDayWed: boolean;
    public routeDayThu: boolean;
    public routeDayFri: boolean;
    public routeDaySat: boolean;
    public routeDaySun: boolean;
    public pricingBoundaryEnabled: boolean;
    public routePricings: Array<routepricing>;

    //Additional Fields
    public pickupCount: number;
    public passengerCount: number;
    public pickups: CustomerPickup[];
    public vehicleId: number;
    public assignedVehicle: string;
    public vehicleCapacity: number;
    public driverId: number;
    public driverName: string;
    public isCancelled: boolean;
    public routeDayId: number;
}

File: Models/Route.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace pathfinderRoute.Models
{
    public class Route
    {
        public int Id { get; set; }
        public string RouteName { get; set; }
        public string RouteDescription { get; set; }
        public DateTime ScheduledDeparture { get; set; }
        public DateTime RouteStarts { get; set; }
        public DateTime? RouteExpires { get; set; }
        public decimal BasePrice { get; set; }
        public string FromAddress { get; set; }

        [Column(TypeName = "decimal(9, 6)")]
        public decimal FromGeoLat { get; set; }

        [Column(TypeName = "decimal(9, 6)")]
        public decimal FromGeoLong { get; set; }
        public string ToAddress { get; set; }

        [Column(TypeName = "decimal(9, 6)")]
        public decimal ToGeoLat { get; set; }

        [Column(TypeName = "decimal(9, 6)")]
        public decimal ToGeoLong { get; set; }
        public bool RouteDayMon { get; set; }
        public bool RouteDayTue { get; set; }
        public bool RouteDayWed { get; set; }
        public bool RouteDayThu { get; set; }
        public bool RouteDayFri { get; set; }
        public bool RouteDaySat { get; set; }
        public bool RouteDaySun { get; set; }
        public bool PricingBoundaryEnabled { get; set; }
        public DateTime Added { get; set; }
        public string AddedBy { get; set; }
        public DateTime Modified { get; set; }
        public string ModifiedBy { get; set; }
        public List<Pickup> Pickups { get; set; }
        //public int? VehicleId { get; set; }
        //public Vehicle Vehicle { get; set; }
        public List<RouteDay> RouteDays { get; set; }
        public List<RoutePricing> RoutePricings { get; set; }
    }
}